ibattery.py

```diff
@@ -0,0 +1,237 @@
+        ),
+        (
+            "foreground",
+            "ffffff",
+            "Battery color in normal mode."
+        ),
+        (
+            "charging_fg",
+            "02a724",
+            "foreground color when battery is charging."
+        ),
+        (
+            "update_interval",
+            30,
+            "time to wait until the widgets refreshes."
+        ),
+        (
+            "low_foreground",
+            "ff0000",
+            "change color when battery is low."
+        ),
+        (
+            "warn_below",
+            10,
+            "battery level to indicate battery is low."
+        ),
+        (
+            "notify",
+            False,
+            "send a notification when battery is low."
+        ),
+        (
+            "notification_timeout",
+            10,
+            "time in seconds to display notification."
+        ),
+    ]
+
+    def __init__(self, **config):
+        self.widget_width = 36
+        base._Widget.__init__(self, bar.CALCULATED, **config)
+        self.add_defaults(Battery.defaults)
+
+        self.length = self.padding * 2 + self.widget_width
+        self.HEIGHT = 16  # widgets height
+        self.BAR_WIDTH = 30  # battery bar
+
+        self._has_notified = False
+        self.timeout = int(self.notification_timeout * 1000)
+
+        self._foreground = self.foreground
+
+    def _notify(self, percent):
+        if not self._has_notified:
+            send_notification(
+                "Warning",
+                f"Battery at {percent}%\ncharge me baby!",
+                urgent=True,
+                timeout=self.timeout
+            )
+            self._has_notified = True
+        else:
+            self._has_notified = False
+
+    def update(self):
+        percent, charging = self.get_bat()
+        if self.notify and percent < self.warn_below:
+            self._notify(percent)
+        self.configure_(percent, charging)
+        self.draw_battery(percent, charging)
+
+    def configure_(self, percent, plugged):
+        if plugged:
+            self.foreground = self.charging_fg
+        elif percent <= self.warn_below:
+            self.foreground = self.low_foreground
+        else:
+            self.foreground = self._foreground
+
+    def calculate_length(self):
+        if self.bar.horizontal:
+            return (self.padding * 2) + self.widget_width
+        else:
+            return 0
+
+    def draw(self):
+        percent, charging = self.get_bat()
+        self.configure_(percent, charging)
+        self.draw_battery(percent, charging)
+
+    def draw_battery(self, percent, charging):
+        self.drawer.clear(self.background or self.bar.background)
+        if self.bar.horizontal:
+            PERCENT = self.BAR_WIDTH / 100 * percent
+            y_margin = (self.bar.height - self.HEIGHT) / 2
+
+            self.drawer.set_source_rgb("8c8c8c")
+            self._fill_body(
+                1,
```

Comment:
I'd put the padding value in here (and adjust the other parts where you draw/fill rectangles etc).